bridge/client/app/_services/data.service.ts

```diff
@@ -197,13 +207,24 @@ export class DataService {
       .pipe(
         map(project => Project.fromJSON(project))
       ).subscribe((project: Project) => {
+      const projects = this._projects.getValue();
+      const existingProject = projects?.find(p => p.projectName === project.projectName);
+      if (existingProject) {
+        const {roots, sequences, ...copyProject} = project;
+        Object.assign(existingProject, copyProject);
+        this._projects.next(projects);
+      }
+    }, err => {
+      if (err.status === 404) {
         const projects = this._projects.getValue();
-        const existingProject = projects?.find(p => p.projectName === project.projectName);
-        if (existingProject){
-          const {roots, sequences, ...copyProject} = project;
-          Object.assign(existingProject, copyProject);
+        const projectIdx = projects?.findIndex(p => p.projectName === projectName);
+        if (projectIdx && projectIdx > 0) {
+          projects?.splice(projectIdx);
```

Comment:
change it to `splice(projectIdx, 1)`.
Also index 0 is excluded, is there a reason why?

Reply:
Changed it. No reason, just a mistake. Glad you saw it.